streaming/src/main/scala/org/apache/spark/streaming/dstream/InputDStream.scala

```diff
@@ -47,6 +49,14 @@ abstract class InputDStream[T: ClassTag] (@transient ssc_ : StreamingContext)
   /** This is an unique identifier for the input stream. */
   val id = ssc.getNewInputStreamId()
 
+  // Keep track of the freshest rate for this stream using the rateEstimator
+  protected[streaming] val rateController: Option[RateController] =
+    RateEstimator.makeEstimator(ssc.conf).map { estimator =>
```

Comment:
And InputDStream should not have a RateController by default. So this should be None. Its will be upto implementation of InputDStream to override and define.